File: ie/AllowedPniNpnList.go
package ie

type AllowedPniNpnList struct {
	AllowedPniNpnItem AllowedPniNpnItem `bitstring:"sizeLB:0,sizeUB:150"`
}

type AllowedPniNpnItem struct {
	PlmnIdentity          PlmnIdentity          `bitstring:"sizeLB:0,sizeUB:150"`
	PniNpnRestricted      []byte                `bitstring:"sizeLB:0,sizeUB:150"`
	AllowedCagListPerPlmn AllowedCagListPerPlmn `bitstring:"sizeLB:0,sizeUB:150"`
}

type AllowedCagListPerPlmn struct {
	CagId CagId `bitstring:"sizeLB:0,sizeUB:150"`
}

File: generator/cache/AllowedPniNpnList.py
ies = []
ies.append({ "iei" : "AllowedPniNpnItem", "level" : "0", "range" : "1..<maxnoofEPLMNs+1>", "type" : ""})
ies.append({ "iei" : "PlmnIdentity", "level" : "1", "range" : "", "type" : "PlmnIdentity"})
ies.append({ "iei" : "PniNpnRestricted", "level" : "1", "range" : "", "type" : "ENUMERATED (restricted, not-restricted, …)"})
ies.append({ "iei" : "AllowedCagListPerPlmn", "level" : "1", "range" : "1..<maxnoofAllowedCAGsperPLMN>", "type" : ""})
ies.append({ "iei" : "CagId", "level" : "2", "range" : "", "type" : "CagId"})
